code/datums/greyscale/_greyscale_config.dm

```diff
@@ -0,0 +1,136 @@
+/datum/greyscale_config
+	/// Reference to the json config file
+	var/json_config
+
+	/// Reference to the dmi file for this config
+	var/icon_file
+
+	///////////////////////////////////////////////////////////////////////////////////////////
+	// Do not set any further vars, the json file specified above is what generates the object
+
+	/// String path to the json file, used for reloading
+	var/string_json_config
+
+	/// String path to the icon file, used for reloading
+	var/string_icon_file
+
+	/// Layer objects that the sprite is made up of
+	var/list/layers
+
+	/// How many colors are expected to be given when building the sprite
+	var/expected_colors = 0
+
+	/// Generated icons keyed by their color arguments
+	var/list/icon_cache
+
+// There's more sanity checking here than normal because this is designed for spriters to work with
+// Sensible error messages that tell you exactly what's wrong is the best way to make this easy to use
+/datum/greyscale_config/New()
+	if(!json_config)
+		CRASH("Greyscale config object [DebugName()] is missing a json configuration, make sure `json_config` has been assigned a value.")
+	string_json_config = "[json_config]"
+	if(!icon_file)
+		CRASH("Greyscale config object [DebugName()] is missing an icon file, make sure `icon_file` has been assigned a value.")
+	string_icon_file = "[icon_file]"
+
+	Refresh()
+
+/datum/greyscale_config/proc/Refresh(loadFromDisk=FALSE)
+	if(loadFromDisk)
+		json_config = file(string_json_config)
+		icon_file = file(string_icon_file)
+
+	var/list/raw = json_decode(file2text(json_config))
+	layers = ReadLayersFromJson(raw["layers"])
+	if(!length(layers))
+		CRASH("The json configuration [DebugName()] is missing any layers.")
+
+	icon_cache = list()
+
+	ReadMetadata()
+
+/// Gets the name used for debug purposes
+/datum/greyscale_config/proc/DebugName()
+	return "([icon_file]|[json_config])"
+
+/// Takes the json layers configuration and puts it into a more processed format
+/datum/greyscale_config/proc/ReadLayersFromJson(list/data)
+	var/list/output = ReadLayerGroup(data)
+	return output[1]
```

Comment:
Is the first item in this list always the json layers config? Why not use a define for the indexing?
hell can't ReadLayerGroup return a non list?

Reply:
Due to the nature of the recursive proc below, anything it returns is put in a list because it's putting lists in other lists. I can shift some logic around to unwrap it elsewhere but doing it here is simple